integration_tests/src/main/python/json_test.py

```diff
@@ -496,16 +496,22 @@ def test_from_json_struct(schema):
             .select(f.from_json('a', schema)),
         conf={"spark.rapids.sql.expression.JsonToStructs": True})
 
-@pytest.mark.xfail(reason='https://github.com/NVIDIA/spark-rapids/issues/9597')
-def test_from_json_struct_boolean():
-    json_string_gen = StringGen(r'{ "a": [truefalsTRUEFALS]{0,5}, "b": [0-9]{0,2} }') \
-        .with_special_pattern('{ "a": true, "b": 1 }', weight=50) \
-        .with_special_pattern('{ "a": false, "b": 0 }', weight=50) \
-        .with_special_pattern('', weight=50) \
-        .with_special_pattern('null', weight=50)
+@pytest.mark.parametrize('pattern', [
+    r'{ "bool": [truefalsTRUEFALS]{1,5} }',
+    r'{ "bool": "[truefalsTRUEFALS]{1,5}" }',
```

Comment:
So what is the probability that this actually shows up as "true" or "false"? I think it has to be worse than `1 out of 524288 rows 2/(16^5)` because that would be the probability for `{5}` at the end, but it is `{1,5}`. Those values should be returned as `null` because a quoted string should not be interpreted as a boolean, but I don't think this code is doing that. I could be wrong though, but the probabilities don't give me a lot of comfort that we have covered that test case.

Reply:
I have updated the tests and added a link to the issue